File: frontend/src/tipos/administracao.ts
// tipos/administracao.ts — Tipos do dashboard administrativo

export interface TendenciaResposta {
  predominante: string;
  abaixo: number;
  acima: number;
  igual: number;
}

export interface ResumoAdministracao {
  participantes_total: number;
  participantes_completos: number;
  tentativas_validas_total: number;
  tentativas_esperadas: number;
  taxa_conclusao_percentual: number;
  erro_medio_ms: number;
  erro_absoluto_medio_ms: number;
  tendencia_geral: TendenciaResposta;
}

export interface EstatisticaItem {
  chave: string;
  quantidade_tentativas: number;
  erro_medio_ms: number;
  erro_absoluto_medio_ms: number;
  abaixo: number;
  acima: number;
  igual: number;
  tendencia_predominante: string;
}

export interface EstatisticasResposta {
  agrupar_por: string;
  itens: EstatisticaItem[];
}

export interface ParticipanteListaItem {
  id: string;
  nome: string;
  codigo: string;
  criado_em: string;
  tentativas_concluidas: number;
  situacao: string;
}

export interface PaginaParticipantes {
  pagina: number;
  tamanho: number;
  total_itens: number;
  total_paginas: number;
  itens: ParticipanteListaItem[];
}

export interface ResumoIndividual {
  erro_medio_ms: number;
  erro_absoluto_medio_ms: number;
  tendencia: TendenciaResposta;
}

export interface ParticipanteAdministracao {
  participante: import('./participante').ParticipanteDetalhe;
  resumo: ResumoIndividual;
}

export interface TokenResposta {
  token_acesso: string;
  tipo_token: string;
  expira_em: string;
}

File: backend/app/esquemas/administracao.py
import uuid
from datetime import datetime

from pydantic import BaseModel

from app.esquemas.participante import ParticipanteDetalheResposta, ParticipanteListaItem


class AcessoAdministracao(BaseModel):
    nome: str
    codigo: str


class TokenResposta(BaseModel):
    token_acesso: str
    tipo_token: str = "bearer"
    expira_em: datetime


class TendenciaResposta(BaseModel):
    predominante: str
    abaixo: int
    acima: int
    igual: int


class ResumoAdministracaoResposta(BaseModel):
    participantes_total: int
    participantes_completos: int
    tentativas_validas_total: int
    tentativas_esperadas: int
    taxa_conclusao_percentual: float
    erro_medio_ms: float
    erro_absoluto_medio_ms: float
    tendencia_geral: TendenciaResposta


class EstatisticaItemResposta(BaseModel):
    chave: str
    quantidade_tentativas: int
    erro_medio_ms: float
    erro_absoluto_medio_ms: float
    abaixo: int
    acima: int
    igual: int
    tendencia_predominante: str


class EstatisticasResposta(BaseModel):
    agrupar_por: str
    itens: list[EstatisticaItemResposta]


class PaginaParticipantesResposta(BaseModel):
    pagina: int
    tamanho: int
    total_itens: int
    total_paginas: int
    itens: list[ParticipanteListaItem]


class ResumoIndividualResposta(BaseModel):
    erro_medio_ms: float
    erro_absoluto_medio_ms: float
    tendencia: TendenciaResposta


class ParticipanteAdministracaoResposta(BaseModel):
    participante: ParticipanteDetalheResposta
    resumo: ResumoIndividualResposta

